modules/standard/IO.chpl

```diff
@@ -2965,14 +3012,33 @@ proc file.readerHelper(param kind=iokind.dynamic, param locking=true,
   on this._home {
     try this.checkAssumingLocal();
     if (region.hasLowBound() && region.hasHighBound()) {
-      ret = new fileReader(kind, locking, this, err, hints, region.low,
-                           region.high, style);
+      if ((fromOpenReader && useNewOpenReaderRegionBounds) ||
+          fromOpenUrlReader ||
+          (!fromOpenReader && !fromOpenUrlReader
+           && useNewFileReaderRegionBounds)) {
+        ret = new fileReader(kind, locking, this, err, hints, region.low,
```

Comment:
This expression feels complex enough that a comment explaining what it's checking might be helpful. Just in case someone has to modify this method before the deprecated behavior gets removed.

Reply:
I've pushed a comment, does it make sense to you?  I also realized that the `!fromOpenUrlReader` portion of it didn't really make sense, so I removed it

Reply:
Yeah that makes sense. Thanks!

Reply:
Thanks!  I'm running through a full paratest now just in case the conditional update broke anything